src/components/TextInput.jsx

```diff
@@ -113,7 +114,7 @@ export default class TextInput extends Component {
     let value = event.target.value;
 
     if (!this.props.allowNewLine && value.match(newLineRegex) !== null) {
-      value = event.target.value.replace(newLineRegex, ' ');
+      value = value.replace(newLineRegex, ' ');
 
```

Comment:
so true...we cannot it needs to behave like controlled component for these features....

we could do that previously with setting domeNode.value, but that was not right approach...I will put some more thoughts on it...